azure-migrate/dependencies-at-scale/AzMig_Dependencies.ps1

```diff
@@ -0,0 +1,698 @@
+                    $row[$header] = $($machine.$header)
+                }
+                $csvData += New-Object PSObject -Property $row
+            }
+            $csvData | Export-Csv -Path $OutputCsvFile -NoTypeInformation -Append
+            Write-Host "List of machines saved to" $OutputCsvFile
+        } 
+        else {
+           Write-Host "No machines discovered in the appliance $appliancename"
+        }            
+    }         
+} 
+
+
+
+
+
+function Set-AzMigDependencyMappingAgentless {
+    [CmdletBinding()]
+    Param(
+        [Parameter(ParameterSetName = 'Enable', Mandatory = $true)]
+        [Parameter(ParameterSetName = 'Disable', Mandatory = $true)]
+        [string]$InputCsvFile,
+
+        [Parameter(ParameterSetName = 'Enable', Mandatory = $true)]
+        [switch]$Enable,
+
+        [Parameter(ParameterSetName = 'Disable', Mandatory = $true)]
+        [switch]$Disable
+    )
+
+    if (-not (Test-path -Path $InputCsvFile)) {
+        throw "File $InputCsvFile not found"
+    }
+    
+	if (-not ($InputCsvFile -imatch ".*\.csv$")) {
+        throw "Input file is not CSV."    
+    }
+	
+    $VMDetails = Import-CSV $InputCsvFile
+	
+	if(-not ($VMDetails[0].psobject.Properties.Name.ToLower().contains("armid")) ) {
+		throw "Input CSV file does not contain required column 'ARMID'"
+	}
+
+    if($Enable){ 
+        $ActionVerb = "Enabled";
+		$EnableDependencyMapping = $true
+    } 
+    elseif ($Disable) {
+        $ActionVerb = "Disabled";
+		$EnableDependencyMapping = $false
+    } 
+    else {
+        throw "Error"
+    }
+    #Checkifthenumberofmachinesexceedthemaximumlimit
+    if ($ActionVerb -eq "Enabled") {
+        $machinesinfo = @{}
+        foreach ($machine in $VMDetails) {
+            $machineid = $machine.ARMID
+            $splitid = $machineid -split '/'
+            $vmwareSitesIndex = $splitid.IndexOf('VMwareSites')
+            $hypervSitesIndex = $splitid.IndexOf('HyperVSites')
+            $serversitesIndex = $splitid.IndexOf('ServerSites')
+
+            if ($vmwareSitesIndex -ne -1 -and $vmwareSitesIndex -lt ($splitid.Count - 1)) {
+                $siteid = ($splitid[0..($vmwareSitesIndex + 1)] -join '/')
+            } 
+            elseif ($hypervSitesIndex -ne -1 -and $hypervSitesIndex -lt ($splitid.Count - 1)) {
+                $siteid = ($splitid[0..($hypervSitesIndex + 1)] -join '/')
+            } 
+            elseif ($serversitesIndex -ne -1 -and $serversitesIndex -lt ($splitid.Count - 1)) {
+                $siteid = ($splitid[0..($serversitesIndex + 1)] -join '/')
+            } 
+            else {
+                throw "Site ID not found in the arm ID."
+            }
+            #storingthecountofmachinespresentincsv
+            if (-not $machinesinfo.ContainsKey($siteid)) {
+                $machinesinfo[$siteid] = @{
+                        'Type' = $null
+                        'Count' = 0
+                        'numberofmachinesthatcanbeenabled' = 0
```

Comment:
Can you think of something to remove this variable. You already have the limit and you already have the machines which are already enabled, if you dont use this variable - numberofmachinesthatcanbeenabled, then how can you use other 2.